### ISSUE TITLE
Missing lombok compiler in pom.xml build section

### ISSUE THREAD

**Issue by hum2u** [spring-petclinic/spring-petclinic-microservices#285]
compilation of submodules with lombok based getter/setter/no arg constructor annotations failed./setter method compilation failures on eclipse 2024-09

the issue was fixed when the following was added to the pm.xml files of such modules

`<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-compiler-plugin</artifactId>
				<configuration>
					<annotationProcessorPaths>
						<path>
							<groupId>org.projectlombok</groupId>
							<artifactId>lombok</artifactId>
							<version>1.18.34</version>
						</path>
					</annotationProcessorPaths>
				</configuration>
			</plugin>
		</plugins>
	</build>`

**Comment by hum2u** [comment]
<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-compiler-plugin</artifactId>
				<configuration>
					<annotationProcessorPaths>
						<path>
							<groupId>org.projectlombok</groupId>
							<artifactId>lombok</artifactId>
							<version>1.18.34</version>
						</path>
					</annotationProcessorPaths>
				</configuration>
			</plugin>
		</plugins>
	</build>

**Comment by AnkushThakur27** [comment]
@hum2u  Can you tell me on which submodules you are getting this error ? I will start working on it asap.

**Comment by hum2u** [comment]
Anything with @Table annotation in source code, specifically
Api-gatewayCustomers-serviceVets-serviceVisits- service




Sent from Yahoo Mail for iPad


On Wednesday, November 13, 2024, 1:48 PM, Ankush Thakur ***@***.***> wrote:




@hum2u Can you tell me on which submodules you are getting this error ? I will start working on it asap.

**Comment by arey** [comment]
Hi @hum2u did you setup Lombok into Eclipse as explained in this article: https://www.baeldung.com/lombok-ide ?

For information, we have an issue to get ride of Lombok: https://github.com/spring-petclinic/spring-petclinic-microservices/issues/251

**Comment by hum2u** [comment]
No. I was not aware of this mechanism.
I naively assumed that the Pom files were all that were needed, and when the code did not compile because of missing generated Lombok getter/setter methods, I stumbled on the fix I mentioned by looking at other github projects that leveraged Lombok.
Speaking for myself, I would prefer to have to do as little ide configuration as possible and have the Pom files take care of the build the app.
I looked at the READMe file, I saw no clues or mentions of the needed ide configuration for the the article you referenced.
Issue is that if your build solution should be independent of build environment. For example I don’t know how you would configure for Lombok in another build environment like Jenkins.
Thanks

Sent from Yahoo Mail for iPhone


On Wednesday, November 13, 2024, 11:14 PM, Antoine Rey ***@***.***> wrote:




Hi @hum2u did you setup Lombok into Eclipse as explained in this article: https://www.baeldung.com/lombok-ide ?

For information, we have an issue to get ride of Lombok: #251
